### PR TITLE
Update release-trueos.sh

### PR DESCRIPTION
**Pull request by beanpole135**
Add a "super_clean_poudriere" step after "clean_poudriere".
This detects/fixes an issue that has appeared on the Trident build server a few times, where poudriere does not clean up the jail "all-the-way", and leaves a dangling dataset/mountpoint/directory without the corresponding configuration files. This causes the jail creation routine in the next step to fail because the mountpoint from the previous job still exists.

### CODE REVIEW

File: release/release-trueos.sh [hunk 1]
```diff
@@ -161,6 +161,23 @@ clean_poudriere()
 	echo -e "y\n" | poudriere ports -d -p ${POUDRIERE_PORTS}
 }
 
+super_clean_poudriere()
+{
+	#Look for any leftover mountpoints/directories and remove them too
+	for i in `ls ${POUDRIERED_DIR}/*-make.conf`
+	do
+		name=`basename "${i}" | rev | cut -d "-" -f 2-10 | rev`
```
Comment (kmoore134) [inline release/release-trueos.sh]: This looks *real* dangerous. If there are more or less '-' characters then something could go wrong here. 

Recommend you cd ${POUDRIERED_DIR} and ls '*-make.conf' instead. Then to get the jail/base name, you can do echo $i | sed 's|-make.conf||g' without that horrible rev / cut / rev stuff ;)

File: release/release-trueos.sh [hunk 2]
```diff
@@ -161,6 +161,23 @@ clean_poudriere()
...
+		if [ ! -d "${POUDRIERED_DIR}/jails/${name}" ]  && [ -d "${POUDRIERE_BASEFS}/jails/${name}" ] ; then
+			#Jail configs missing, but jail mountpoint still exists
+			#Need to completely destroy the old jail dataset/dir
+			_stale_dataset=$(mount | grep 'on ${POUDRIERE_BASEFS}/jails/${name} ' | cut -w -f 1)
+			umount ${POUDRIERE_BASEFS}/jails/${name}
+			rmdir ${POUDRIERE_BASEFS}/jails/${name}
+			zfs destroy -r ${_stale_dataset}
```
Comment (kmoore134) [inline release/release-trueos.sh]: Need some error checking here. You probably want to validate _stale_dataset exists or isn't bogus before you start running zfs destroy.

**Review by kmoore134 — CHANGES_REQUESTED** [review]

File: release/release-trueos.sh (hunk 1 above)
Comment (kmoore134) [reply release/release-trueos.sh]: Nevermind on the "cd" part, didn't see the 'basename' section.

Comment (beanpole135) [reply release/release-trueos.sh]: Done!

File: release/release-trueos.sh (hunk 2 above)
Comment (beanpole135) [reply release/release-trueos.sh]: done!

**Review by kmoore134 — APPROVED** [review]

**Pull request merged**